packages/playwright-test/src/mount.ts

```diff
@@ -48,23 +48,27 @@ export const fixtures: Fixtures<
     },
 
     mount: async ({ page }, use) => {
-      await use(async (component: JsxComponent | string, options?: MountOptions) => {
-        const selector = await (page as any)._wrapApiCall(async () => {
-          return await innerMount(page, component, options);
-        }, true);
-        const locator = page.locator(selector);
-        return Object.assign(locator, {
-          unmount: async () => {
-            await locator.evaluate(async () => {
-              const rootElement = document.getElementById('root')!;
-              await window.playwrightUnmount(rootElement);
-            });
-          },
-          update: async (options: JsxComponent | Omit<MountOptions, 'hooksConfig'>) => {
-            if (isJsxApi(options)) return await innerUpdate(page, options);
-            await innerUpdate(page, component, options);
-          }
-        });
+      await use((component: JsxComponent | string, options?: MountOptions) => {
+        async function mount() {
+          const selector = await (page as any)._wrapApiCall(async () => {
+            return await innerMount(page, component, options);
+          }, true);
+          const locator = page.locator(selector);
+          return Object.assign(locator, {
+            unmount: async () => {
+              await locator.evaluate(async () => {
+                const rootElement = document.getElementById('root')!;
+                await window.playwrightUnmount(rootElement);
+              });
+            },
+            update: async (options: JsxComponent | Omit<MountOptions, 'hooksConfig'>) => {
+              if (isJsxApi(options)) return await innerUpdate(page, options);
+              await innerUpdate(page, component, options);
+            }
+          });
+        }
+        page.addListener('load', mount);
```

Comment:
This change suggests that it is Ok to navigate in the component tests. But in reality, only 'good' navigations to a magic URL will result in the harness result (the one with window.playwrightMount available. If we want to support it, we should ensure that the page navigated to any URL allows mounting?

Reply:
> If we want to support

I have some router logic which i think is useful to test with `page.goto` and `page.refresh`: https://github.com/sand4rt/vue-router-test/blob/master/src/router/index.ts#L6

Also wrote some tests for this logic: https://github.com/sand4rt/vue-router-test/blob/master/src/notation-vue.test.ts#L4

Wonder what you think and if there is a better way to test this logic?

> We should ensure that the page navigated to any URL allows mounting?

When testing components, navigating to pages that don't start with the base URL `process.env.PLAYWRIGHT_TEST_BASE_URL` seems not useful to me. A URL that starts with the base URL but doesn't exist/is not registered in [`router/index.ts`](https://github.com/sand4rt/vue-router-test/blob/master/src/router/index.ts#L8), like: `http://localhost:3100/page-that-does-not-exists` is in my opinion useful to keep.